cpp/cmake/Modules/ConfigureGoogleTest.cmake

```diff
@@ -0,0 +1,49 @@
+set(GTEST_ROOT "${CMAKE_BINARY_DIR}/googletest")
```

Comment:
Similar to google benchmark above, this is a hack to build at configuration time which isn't a good idea. Should similarly make this a `ExternalProject_add`.